documentation improvements, slight code changes to accommodate hooks module

jhallard opened the pull request:
The documentation has been updated slightly to be more clean and explicit. I also added the `--hook_data_json` argv item to allow a user to pass in arbitrary data through to the hook module. Simply submit a valid json dictionary to this argument, we will read it into a python dictionary and forward that to the `set_hook_data()` function inside the hook module (should it exist) before we do anything else with the module. The logger is also passed in so that the hook module can use the same logger as the importer script.

jhallard committed:
updating the README slightly

jhallard committed:
Removed the early return statements, jobs-plugin is now live

jhallard committed:
Improving documenation, added means of getting arbitrary data to hooks

We now have a `--hook_data_json` argument that takes in a json
dictionary of values, deserializes it into a python dictionary and we
pass that dictionary along to the `set_hook_data()` function in the
hook.

jhallard committed:
Moved all print statements to use dedicated logger

Also added argv defines for setting the logger to verbose and debug

jhallard committed:
More print statement removals, general cleaning up of the code

jhallard committed:
Merge branch 'john-fix-import' into john-add-docs

john-fix-import branch had an open PR but was getting left behind by
some of my other changes, so I took what was good from that PR and put
it in this branch. I'll close the old PR and open a new one with all of
my newest changes.

jcmaslan commented on README.md:
```diff
@@ -23,28 +23,41 @@ which tuna or marlin was brought on board as in [this demo](https://www.youtube.
-  -i HOST, --host HOST  the segmenter's IP address
-  -p PORT, --port PORT  the segmenter's port number
+                        folder to process
+  -i HOST, --host HOST  the segmenter ip
```
@jhallard is this limited to IP address or can it also be a host name on the local network network?

jhallard replied:
It does not have to be an IP address, good point. Will update to reflect such.

jhallard committed:
Slight logging improvements, mostly cleaning up

jcmaslan commented on README.md:
```diff
@@ -23,28 +23,41 @@ which tuna or marlin was brought on board as in [this demo](https://www.youtube.
+                        full path to hook module for custom functions (a
+                        python file)
+  -d HOOK_DATA_JSON, --hook_data_json HOOK_DATA_JSON
+                        a json-dictionary containing extra information to be
```
Is it necessarily a dictionary? Isn't it just an JSON Object?

jhallard replied:
you're right, my terminology is mixed up. It is just an object.

jcmaslan commented on README.md:
```diff
@@ -135,11 +147,36 @@ def post_video_content(host, port, camera_name, camera_id, filepath, timestamp,
         filepath    - full path to the video file that needs segmentation
         timestamp   - the earliest timestamp contained in the video file
-        latlng (opt) - the lat/long of the camera (as parsed from the filename)
+        location (opt) - a string describing the location of the camera (example lat-long)
```
please make example concrete.

jhallard replied:
where did my example go? It was here... will fix with concrete example

jhallard replied:
got it, the example was: 

```json
{"location": {"lat": 7.367598, "lng":134.706975}, "accuracy":5.0}
```

mdipierro replied:
oops. sorry. Perhaps I made a merge error.

jcmaslan commented on README.md:
```diff
@@ -135,11 +147,36 @@ def post_video_content(host, port, camera_name, camera_id, filepath, timestamp,
+
+#### Set Hook Data Function
+
+Sometimes there is extra data that the hook-module needs from the user but is not explicitely defined in the importer arguments,
```
explicitly

jcmaslan commented on README.md:
````diff
@@ -135,11 +147,36 @@ def post_video_content(host, port, camera_name, camera_id, filepath, timestamp,
+python importer "... some arguments here..." --hook_data_json '{"camera_plan": "pro", "user_id": "AABBCCDD"}'
+```
+
+And the json-dictionary that you define will be de-serialized into a python dictionary and passed to the `set_hook_data` function defined
````
Same question about dictionary vs. any object at all?

jcmaslan commented on importer.py:
```diff
@@ -40,11 +41,11 @@ def get_params(self, path):
                 except: pass
         if not camera:
-            print 'did not detect camera name, assuming "default"'
+            logging.warn('did not detect camera name, assuming "default"')
```
If this is a name placeholder, how about "Unnamed" instead so convey that it's not related to a system default.

jcmaslan reviewed (approved):
LGTM with changes requested

jhallard committed:
Slight changes to documentation as per @jcmaslan requests

jhallard committed:
Some more documenation changes

mdipierro commented:
LGTM

jhallard committed:
Merge branch 'master' into john-add-docs

jhallard committed:
Slight fixes, add logging on error

jhallard committed:
Continued documenation improvements

jhallard committed:
Adding camera-object tracking, improved error handling

jhallard committed:
Added support for camera-config tracking

We now keep track of the config returned for each camera regitered and
pass this along to the hook module for it to be used

jhallard committed:
Making meta-data extraction from movies safe

jhallard committed:
Track if all video files are already uploaded, exit with success message

jhallard committed:
Added code to handle tracking the job-id and returning that to the user

The user can then submit this job-id to a batch-download script to get
the definition of the shards that compose this job and use that info to
assemble the collection of labels

jhallard committed:
Merge branch 'master' into john-add-docs

changed name from importer.py to import_video.py (to avoid pypi
conflict), documenation has been updated accordingly.

jhallard committed:
Changed setup.py to follow new naming convention

jhallard committed:
Setting description and dependencies in setup script

jhallard committed:
removed csv dependency (turns out it's built-in) and tested setup script

spoiler-alert: it works :)
